### ISSUE TITLE
Make it so that when you `git-ps rr` it also opens create PR url in default browser

### ISSUE THREAD

**Issue by drewdeponte** [uptech/git-ps#12]
This is a half step improvement that would be easy to implement and streamline the workflow a tiny bid more.

Ideally if we were doing a GUI version of this we would do the oauth integration with GitHub/Bitbucket, etc. to facilitate actually interacting with their APIs to create the pull request.

**Comment by drewdeponte** [comment]
Also, GitHub CLI recently was announced. Maybe there is an option to integrate with that to actually create the pull request somehow.

https://thenewstack.io/introducing-the-github-cli/

**Comment by brandonpittman** [comment]
1. Check for `.github` directory
2. `gh repo view --web`

That could work.

**Comment by drewdeponte** [comment]
Yeah, something like that is a possibility. Part of why I have been holding off on something like this is because it isn't so much git specific as it is service specific (GitHub Cloud, GitHub Enterprise, Bitbucket Cloud, Bitbucket Server, GitLab, etc.).

Avoiding the specific services is really nice from a maintainability standpoint.

**Comment by brandonpittman** [comment]
@drewdeponte What about a hooks setup that users could run shell scripts from?

```fish
# .pshooks/post-rr

if test -d .github
	gh repo view --web
end
```

**Comment by drewdeponte** [comment]
@brandonpittman that is actually a fantastic idea and something I think should be prioritized.

**Comment by jazzdragon** [comment]
@drewdeponte another approach might be to create extensions, managed in separate repos, that users can install for a specific service. Something like `git-ps-github`. Advantages that come to mind over hooks:
- Less work for the user to implement.
- Opens more possibilities other than opening a PR automatically, such as displaying in `git ps ls` if a PR was approved, rejected, etc.
- No added overhead for maintaining this repo.

These ideas are actually not mutually exclusive...

Thoughts?

**Comment by drewdeponte** [comment]
I am not sure about a specific extension like `git-ps-github`. The hooks support is much more generic and would give the users other the option to whip up custom scripts. But we could still have ease of install by providing a wiki page or something that tracks community provided hooks.

I guess extension would in theory be generic as well if I provided some API for those extensions to integrate. But I think there is more overhead in that than whipping up a hook.

Definitely open to thoughts.

**Comment by jazzdragon** [comment]
I guess the hook should be implemented whether we create extensions or not.
Do you mind if I take a stab at it?

**Comment by drewdeponte** [comment]
Not all. I welcome contributions.

The only thing is that I have halted development on git-ps as I have started development on the next major release of git-ps which is written in Rust and uses libgit2 which has many many benefits. The code base and development of that are happening in https://github.com/uptech/git-ps-rs

So if anything I think we should add the functionality over there.

But there is other functionality that still needs to be implemented there for it to have parity with the swift version.

But I am happy for any help I can get in developing it.
